cpp/core/compute/Backend.h

```diff
@@ -118,7 +118,66 @@ class Backend : public std::enable_shared_from_this<Backend> {
   std::unordered_map<std::string, std::string> confMap_;
 };
 
-void setBackendFactory(std::function<std::shared_ptr<Backend>()> factory);
+using BackendFactory1 = std::shared_ptr<Backend> (*)(const std::unordered_map<std::string, std::string>&);
```

Comment:
The names `BackendFactory1` and `BackendFactory2` look some kind of ambiguous to me. Can we rename `BackendFactory1` as `BackendFactoryWithConf`?

Reply:
OK